From 354e2749ca0ed5fe2d2241b68d9ee77327e7699c Mon Sep 17 00:00:00 2001
From: Martin Kosek <mkosek@redhat.com>
Date: Fri, 13 Jun 2014 12:17:40 +0200
Subject: [PATCH 108/116] Winsync re-initialize should not run memberOf fixup
 task

Change re-initialize command to consider memberOf fixup task only
for non-winsync replication agreements.

https://fedorahosted.org/freeipa/ticket/3854
---
 install/tools/ipa-replica-manage | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/install/tools/ipa-replica-manage b/install/tools/ipa-replica-manage
index 9e5cbf5218365fb8dee488ed794f915e7db5eab7..914f8aec4e3a7052b845007d6f8567427602eff8 100755
--- a/install/tools/ipa-replica-manage
+++ b/install/tools/ipa-replica-manage
@@ -740,11 +740,11 @@ def re_initialize(realm, thishost, fromhost, dirman_passwd):
         repl.initialize_replication(agreement.dn, repl.conn)
         repl.wait_for_repl_init(repl.conn, agreement.dn)
 
-    # If the agreement doesn't have nsDS5ReplicatedAttributeListTotal it means
-    # we did not replicate memberOf, do so now.
-    if not agreement.getValue('nsDS5ReplicatedAttributeListTotal'):
-        ds = dsinstance.DsInstance(realm_name = realm, dm_password = dirman_passwd)
-        ds.init_memberof()
+        # If the agreement doesn't have nsDS5ReplicatedAttributeListTotal it means
+        # we did not replicate memberOf, do so now.
+        if not agreement.getValue('nsDS5ReplicatedAttributeListTotal'):
+            ds = dsinstance.DsInstance(realm_name = realm, dm_password = dirman_passwd)
+            ds.init_memberof()
 
 def force_sync(realm, thishost, fromhost, dirman_passwd):
 
-- 
1.9.3

